lib/astarte_data_access/config.ex

```diff
@@ -38,4 +57,139 @@ defmodule Astarte.DataAccess.Config do
+  app_env :pool_size, :astarte_data_access, :pool_size,
+    os_env: "CASSANDRA_POOL_SIZE",
+    type: :integer,
+    default: 1
+
+  @envdoc "Enable SSL for Cassandra connections. Defaults to false."
+  app_env :ssl_enabled, :astarte_data_access, :ssl_enabled,
+    os_env: "CASSANDRA_SSL_ENABLED",
+    type: :boolean,
+    default: false
+
+  @envdoc "Disable Server Name Indication. Defaults to false."
+  app_env :ssl_disable_sni, :astarte_data_access, :ssl_disable_sni,
+    os_env: "CASSANDRA_SSL_DISABLE_SNI",
+    type: :boolean,
+    default: false
+
+  @envdoc """
+  Specify the hostname to be used in TLS Server Name Indication extension.
+  If not specified, the cassandra nodes will be used. This value is used
+  only if Server Name Indication is enabled.
+  """
+  app_env :ssl_custom_sni, :astarte_data_access, :ssl_custom_sni,
+    os_env: "CASSANDRA_SSL_CUSTOM_SNI",
+    type: :binary
+
+  @envdoc """
+  Specifies the certificates of the root Certificate Authorities to be trusted.
+  When not specified, the bundled cURL certificate bundle will be used.
+  """
+  app_env :ssl_ca_file, :astarte_data_access, :ssl_ca_file,
+    os_env: "CASSANDRA_SSL_CA_FILE",
+    type: :binary,
+    default: CAStore.file_path()
+
+  @envdoc """
+  Discover nodes in the same cluster as specified in CASSANDRA_NODES. If your Cassandra
+  instance is outside of your network, enabling the autodiscovery leads to connection failures.
+  Defaults to false.
+  """
+  app_env :autodiscovery_enabled, :astarte_data_access, :autodiscovery_enabled,
+    os_env: "CASSANDRA_AUTODISCOVERY_ENABLED",
+    type: :boolean,
+    default: false
+
+  defp populate_xandra_ssl_options(options) do
+    if ssl_enabled!() do
+      ssl_options = build_ssl_options()
+      Keyword.put(options, :transport_options, ssl_options)
+    else
+      options
+    end
+  end
+
+  defp populate_cqex_ssl_options(options) do
+    if ssl_enabled!() do
+      ssl_options = build_ssl_options()
+      Keyword.put(options, :ssl, ssl_options)
+    else
+      options
+    end
+  end
+
+  defp build_ssl_options do
+    [
+      cacertfile: ssl_ca_file!(),
+      verify: :verify_peer,
+      server_name_indication: :disable
+    ]
+  end
+
+  defp xandra_authentication_options! do
+    {Xandra.Authenticator.Password,
+     [
+       username: cassandra_username!(),
+       password: cassandra_password!()
+     ]}
+  end
+
+  defp cqex_authentication_options! do
+    {
+      :cqerl_auth_plain_handler,
+      [{cassandra_username!(), cassandra_password!()}]
+    }
+  end
+
+  @spec xandra_options!() :: [xandra_opts]
+  def xandra_options! do
+    [
+      nodes: xandra_nodes!(),
+      authentication: xandra_authentication_options!(),
+      pool_size: pool_size!(),
+      encryption: ssl_enabled!(),
+      autodiscovery: autodiscovery_enabled!()
+    ]
+    |> populate_xandra_ssl_options()
+  end
+
+  @spec xandra_options!(name: String.t()) :: [xandra_opts]
+  def xandra_options!(name: name) do
```

Comment:
Same goes here: just leave `xandra_options!/0` and add the name where you call it